### PR TITLE
Increase maximum hook length to 191 characters

### LINKED ISSUE #16169: An error occurs when trying to register a hook whose name is longer than 64 characters
**Describe the bug**
Error!
Exception thrown by module on reset. Error when resetting module. The length of property Hook->name is currently 65 chars. It must be between 0 and 64 chars.

**To Reproduce**
Hook registration in the module.
$this->registerHook('actionBeforeAjaxDieOrderControllerdisplayAjaxselectDeliveryOption')

**Screenshots**
image

**Additional information**
PrestaShop version: 1.7.6.0
PHP version: 7.1

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Hook names were unnecessarily fixed at a maximum of 64 characters, which could be especially problematic for hooks whose names are dynamically generated (see fixed issue). This change simply increases that limit to 191 (maximum allowed by utf8mb4).
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #16169
| How to test?  | Install or update PS, then install a module that simply registers a hook like this: `$this->registerHook('actionBeforeAjaxDieOrderControllerdisplayAjaxselectDeliveryOption')`

This PR also fixes an invalid SQL statement in the update script.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16176)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

File: classes/Hook.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ class HookCore extends ObjectModel
         'table' => 'hook',
         'primary' => 'id_hook',
         'fields' => array(
-            'name' => array('type' => self::TYPE_STRING, 'validate' => 'isHookName', 'required' => true, 'size' => 64),
+            'name' => array('type' => self::TYPE_STRING, 'validate' => 'isHookName', 'required' => true, 'size' => 255),
```
Comment (PierreRambaud) [inline classes/Hook.php]:
```suggestion
            'name' => array('type' => self::TYPE_STRING, 'validate' => 'isHookName', 'required' => true, 'size' => 191),
```
For utf8mb4 :roll_eyes:

Comment (eternoendless) [reply classes/Hook.php]: I don't suppose anyone in their right mind would use emojis + long hook names?!

Comment (PierreRambaud) [reply classes/Hook.php]: People like using emojis everywhere right now :(

Comment (eternoendless) [reply classes/Hook.php]: Ok as far as I can tell the 191 limit applies for indexed columns regardless of their content, so I'll have to change it.

**Review by PierreRambaud — APPROVED** [review]

**Comment by sarahdib** [comment]
@eternoendless there is an error when I try to re-install the module with a long hook. 
Capture d’écran 2019-11-06 à 16 44 33

**Comment by PierreRambaud** [comment]
Thanks @eternoendless

**Pull request merged**